SonarQube.TeamBuild.PreProcessor/PreprocessorObjectFactory.cs

```diff
@@ -0,0 +1,61 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="PreprocessorObjectFactory.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using SonarQube.Common;
+using System;
+
+namespace SonarQube.TeamBuild.PreProcessor
+{
+    internal class PreprocessorObjectFactory : IPreprocessorObjectFactory
+    {
+        private static readonly IPreprocessorObjectFactory instance = new PreprocessorObjectFactory(); 
+
+        public static IPreprocessorObjectFactory Instance {  get { return instance; } }
```

Comment:
I added tests for this class in a subsequent commit when I amended the interface to support creating the embedded analyser installer.